distributed/stealing.py

```diff
@@ -186,6 +191,54 @@ def move_task(self, key, victim, thief):
                 pdb.set_trace()
             raise
 
+    def move_task_confirm(self, key=None, worker=None, state=None):
+        try:
+            d = self.in_flight.pop(key)
+            thief = d['thief']
+            victim = d['victim']
+            if self.scheduler.task_state.get(key) != 'processing':
+                self.scheduler.occupancy[thief] = sum(self.scheduler.processing[thief].values())
+                self.scheduler.occupancy[victim] = sum(self.scheduler.processing[victim].values())
+                return
+
+            # One of the pair has left, punt and reschedule
+            if (thief not in self.scheduler.workers or
+                victim not in self.scheduler.workers):
+                self.scheduler.reschedule(key)
+                return
+
+            # Victim had already started execution, reverse stealing
+            if state in ('memory', 'executing', 'long-running'):
+                self.scheduler.occupancy[thief] -= d['thief_duration']
+                self.scheduler.total_occupancy -= d['thief_duration']
+                self.scheduler.occupancy[victim] += d['victim_duration']
+                self.scheduler.total_occupancy += d['victim_duration']
+                self.log.append(('already-computing', key, victim, thief))
+                self.scheduler.check_idle_saturated(thief)
+                self.scheduler.check_idle_saturated(victim)
+
+            # Victim was waiting, has given up task, enact steal
+            elif state in ('waiting', 'ready'):
+                self.scheduler.rprocessing[key] = thief
+                del self.scheduler.processing[victim][key]
+                self.scheduler.processing[thief][key] = d['thief_duration']
+                self.remove_key_from_stealable(key)
+                self.put_key_in_stealable(key)
```

Comment:
Do you really mean to `remove_key_from_stealable` and then immediately `put_key_in_stealable`?

Reply:
Empirically seems to make no difference, but the logic escapes me